index.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+const differenceWith = require('lodash/differenceWith');
+const isMatch = require('lodash/isMatch');
+const isPlainObject = require('lodash/isPlainObject');
+
+const waitForActions = (store, actions, timeout) => {
+    timeout = timeout || 50;
+
+    if (typeof actions === 'string' || isPlainObject(actions)) {
+        actions = [actions];
+    }
+
+    if (Array.isArray(actions)) {
+        actions = actions.map((value) => typeof value === 'string' ? { type: value } : value);
+    }
+
+    const shouldPromiseResolve = typeof actions === 'function' ?
+        actions :
+        (storeActions) => differenceWith(actions, storeActions, (action, storeAction) => isMatch(storeAction, action)).length === 0;
```

Comment:
Could we create a function before `waitForActions` with this logic and call it defaultPredicate or something?